ci-merge : Adding option to bypass topic branch in case of merge conflicts and adding a log-parser

sgaud-quic opened the pull request:
This is intended to add functionality in automerge to capture topic branch SHA that are getting merged into the baseline. This will generate a file qcom-next/topic_SHA1 which will have topic brancg name and the SHA that is merged in baseline while running automerge tool.

sgaud-quic committed:
ci-merge : Bypass in case of merge conflict

Introducing a option -p, which will bypass a conflicting branch
in case of non-interactive merge.
Branch would be reset with git merge --abort and next branch
would be merged, instead of exiting.

Signed-off-by: Salendarsingh Gaud <sgaud@qti.qualcomm.com>

shashim-quic commented on ci-merge:
```diff
@@ -618,7 +618,9 @@ echo "Create a new integration branch based on $TAG"
 echo "Merging topic branches..."
-
+mkdir qcom-next
+printf "%*s %*s\n" 10 "Name" 20 "SHA1" > qcom-next/topic_SHA1
```
Let's not redirect o/p to hardcoded file from within code. You can print info here, and then pass to some script to generate o/p in a preferred format that goes into topic_SHA1 file.

sgaud-quic replied:
Updated

shashim-quic commented on ci-merge:
```diff
@@ -663,6 +665,13 @@ while read LINE; do
+    if echo "$MERGE_OUTPUT" | grep -q "Already up to date." ; then
+        echo "Nothing to merge: Already up to date."
+    else
+        printf "%-20s %*s\n" "$REMOTE_NAME" 45 "$branch_tip" >> qcom-next/topic_SHA1
```
same here

sgaud-quic replied:
Updated

shashim-quic commented on ci-merge:
```diff
@@ -644,8 +646,8 @@ while read LINE; do
-
-    git merge -q --no-ff --no-edit -m "Merge remote-tracking branch $REMOTE_NAME into $INTEG_BRANCH" $MERGER
+    branch_tip=$(git rev-parse $MERGER)
+    MERGE_OUTPUT=$(git merge --no-ff --no-edit -m "Merge remote-tracking branch $REMOTE_NAME into $INTEG_BRANCH" $MERGER 2>&1)
```
is 2>&1 necessary ? you can work with all err + success output log ?

sgaud-quic replied:
Updated

sgaud-quic committed:
log-parser : Adding script to parse logs

Script to parse the merge logs, this will search for
"Merge sucessful: " string and capture branch name and SHA
in a text file.

Signed-off-by: Salendarsingh Gaud <sgaud@qti.qualcomm.com>

sgaud-quic committed:
ci-merge : Capture number of commits in merged branch

Calculate and print number of commits present in each branch
that is merged into baseline.

Signed-off-by: Salendarsingh Gaud <sgaud@qti.qualcomm.com>

sgaud-quic committed:
log-parser : Capture number of commits in topic_SHA1

Parse number of commits from log file and add to topic_SHA1 file

Signed-off-by: Salendarsingh Gaud <sgaud@qti.qualcomm.com>

jasonpettit commented:
Can this PR be closed?
